serve/mlc_serve/engine/model_module.py

```diff
@@ -170,7 +170,8 @@ def generate(
 class Tokenizer(Protocol):
     _tokenizer: Any
     eos_token_id: int
-    skip_special_tokens: bool
+    skip_special_tokens: bool # for decoder
+    add_special_tokens: bool # for encoder
```

Comment:
Does `encoder` mean the conversion from string to the token ids? can you elaborate? Also, would be good to add what "special tokens" are in general.

Reply:
Sure, thanks for the comment! encoder decoder here actually means where these options would be used. For example, `skip_special_tokens` would be used in the `decode` function below which process a list of tokein ids to a string, and `add_special_tokens` option would be used in the `encode` function below which process the text string to a list of ids.

I'll note in the docs and add some examples.